boards/01space/esp32c3_042_oled/doc/index.rst

```diff
@@ -0,0 +1,172 @@
+.. _01space_esp32c3_042_oled:
+
+ESP32C3 0.42 OLED
+#################
+
+Overview
+********
+
+ESP32C3 0.42 OLED is a mini development board based on the `Espressif ESP32-C3`_
+RISC-V WiFi/Bluetooth dual-mode chip.
+
+For more details see the `01space ESP32C3 0.42 OLED`_ Github repo.
+
+.. figure:: img/esp32c3_042_oled.webp
+   :align: center
+   :alt: 01space ESP32C3 0.42 OLED
+
+   01space ESP32C3 0.42 OLED
+
+Hardware
+********
+
+This board is based on the ESP32-C3-FH4 with WiFi and BLE support.
+It features:
+
+* RISC-V SoC @ 160MHz with 4MB flash and 400kB RAM
+* WS2812B RGB serial LED
+* 0.42-inch OLED over I2C
+* Qwiic I2C connector
+* One pushbutton
+* Onboard ceramic chip antenna
+* On-chip USB-UART converter
+
+.. note::
+
+   The RGB led is not supported on this Zephyr board yet.
```

Comment:
is there anything special about this LED? i.e. wouldn't something like this work, https://github.com/zephyrproject-rtos/zephyr/blob/bbf19f4f55bf1d87d0a7a2978d137fc378386f03/samples/drivers/led_strip/boards/esp32c3_devkitm.overlay#L13-L29

Reply:
It would. But if I do that then I have to change the pinctrl of spim2 to output MOSI on GPIO2, as the ESP32C3 only has one SPIM peripheral.
And then the nice pinout image in the docs is not correct anymore.

Zephyr only has the bitbanged serial led driver for Cortex-M0 unfortunately.